sunpy/util/decorators.py

```diff
@@ -221,3 +223,43 @@ def __call__(self, func):
         if self.kwargs:
             func.__doc__ = func.__doc__.format(**self.kwargs)
         return func
+
+def deprecate_positional_args(f):
+    """Decorator for methods that issues warnings for positional arguments
+    Using the keyword-only argument syntax in pep 3102, arguments after the
+    * will issue a warning when passed as a positional argument.
+    Parameters
+    ----------
+    f : function
+        function to check arguments on
+
+    References
+    ----------
+    This has been adopted from Scikit-Learn `~scikit-learn/sklearn/utils/validation.py`
+    The Scikit-Learn licence is ~licenses/SCIKIT-LEARN.rst
+    """
+    sig = signature(f)
+    kwonly_args = []
+    all_args = []
+
+    for name, param in sig.parameters.items():
+        if param.kind == Parameter.POSITIONAL_OR_KEYWORD:
+            all_args.append(name)
+        elif param.kind == Parameter.KEYWORD_ONLY:
+            kwonly_args.append(name)
+
+    @wraps(f)
+    def inner_f(*args, **kwargs):
+        extra_args = len(args) - len(all_args)
+        if extra_args > 0:
+            # ignore first 'self' argument for instance methods
+            args_msg = ['{}={}'.format(name, arg)
+                        for name, arg in zip(kwonly_args[:extra_args],
+                                             args[-extra_args:])]
+            warnings.warn("Pass {} as keyword args. From version 2.0 "
```

Comment:
```suggestion
            warnings.warn("Pass {} as keyword args. From version 2.1 "
```
Should this be 2.1 or 3.0?

Reply:
This should probably be passed in as an argument, we could even be smart and use the function in #3982 to calculate it from the deprecated version.

Reply:
May I go forward with using the mentioned function?

Reply:
It was merged in so you can use it now.